Subject: [PATCH] LightyTest.pm: do not shutdown testing socket

While shutting down the sending side of a socket after having
sent the last of the data should not interfere with any future
reception of new data on the same socket, on darwin it does.

If the socket is shutdown for sending, it will fail to receive
any future data from the other end as well unless it already
just happens to be in the receive buffer waiting to be read.

Prevent this problem by skipping the socket shutdown which
allows the tests to run to normal completion and pass.

It's unclear whether or not this problem is limited only to
Perl or is system wide.

Signed-off-by: Kyle J. McKay <mackyle@gmail.com>

---
 tests/LightyTest.pm | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/LightyTest.pm b/tests/LightyTest.pm
index d8850a37..e78fe198 100644
--- a/tests/LightyTest.pm
+++ b/tests/LightyTest.pm
@@ -223,7 +223,7 @@ sub handle_http {
 			print $remote $_.$BLANK;
 			diag("\n<< ".$_) if $is_debug;
 		}
-		shutdown($remote, 1); # I've stopped writing data
+		#shutdown($remote, 1); # I've stopped writing data
 	} else {
 		diag("\nsending request header to ".$host.":".$self->{PORT}) if $is_debug;
 		foreach(@request) {
---
